### ISSUE TITLE
Automation of the build process

### ISSUE THREAD

**Issue by alexanderbock** [OpenSpace/OpenSpace#185]
Setting up a Jenkins job to automatically build and package binaries for Windows, Linux, and Mac

**Comment by alexanderbock** [comment]
0. Create new Jenkins builds (at least on Windows and Mac) that run when code is pushed to `master` or `develop` branch
0. Have the Jenkins job
  a. Build OpenSpace + Launcher
  a. Run the Launcher to download all the files
  a. Package the executable
    i. One version with no data that requires the Launcher to be executed by the user
    i. One version with minimal data that can run the `default.scene` file
    i. One version with all the data as if the Launcher was executed with synchrionizing all `.scene` files
 a. Copy the three versions to the webserver
 a. For `master` builds: Automatically placing it on the openspaceproject.com prebuild webpage
 a. For `develop` builds: Uploading it to the webpage, but publishing the URL in a `deploy` Slack channel

**Comment by rkiddy** [comment]
Is anything happening with this? I am trying to build (just build, not automatedly, just at all) and am so far not successful. As I figure out what I am doing wrong, I want to document the current procedure. An automated build would suggest that build is working without kludges, so I am checking here.

**Comment by rkiddy** [comment]
I did check the page. I tried to do what it suggested. I will document the results.

I had not seen the deploy.bat file in the support directory. Neither of these files mentioned it:
```
https://docs.openspaceproject.com/releases-v0.20/contribute/development/compiling/index.html
https://docs.openspaceproject.com/releases-v0.20/contribute/development/compiling/ubuntu.html
```

I can take this to a seperate issue/PR, if that would be helpful.

**Comment by alexanderbock** [comment]
Feel free to post in the linux channel on Slack and we can take it from there.    There `deploy.bat` only works on Windows, which is why it's not mentioned in those files.

**Comment by alexanderbock** [comment]
A nightly build for Windows is now available at http://data.openspaceproject.com/release/nightly/OpenSpace-minimal.zip

**Comment by rkiddy** [comment]
Is there a Linux and a Mac nightly? Is this ticket done? Or just done for Windoze? Should there be another ticket(s) for the other platforms?

**Comment by alexanderbock** [comment]
Yea, we have an internal ticket for making the Linux packaging automated.  But we won't do the Mac build automated

**Comment by rkiddy** [comment]
Well, I am glad there is a ticket. If it is internal, that means I cannot set a watch for it. Unless there is some way to know about progress? I guess I can just try to remember to check the @linux channel on slack....

**Comment by alexanderbock** [comment]
You're right. Internal tickets suck: https://github.com/OpenSpace/OpenSpace/issues/3763
